mintscript/src/translate.rs

```diff
@@ -321,3 +341,114 @@ impl<C: InputInfoProvider> TranslateInput<C> for TimelockOnly {
+                                            sig.sighash_type(),
+                                            raw_signature,
+                                        )),
+                                    )
+                                }
+                            }
+                        }
+                    };
+                    Ok(script)
+                }
+                TxOutput::IssueNft(_id, _issuance, dest) => Ok(checksig(dest)),
+                TxOutput::DelegateStaking(_amount, _deleg_id) => Err(TranslationError::Unspendable),
+                TxOutput::CreateDelegationId(_dest, _pool_id) => Err(TranslationError::Unspendable),
+                TxOutput::IssueFungibleToken(_issuance) => Err(TranslationError::Unspendable),
+                TxOutput::Burn(_val) => Err(TranslationError::Unspendable),
+                TxOutput::DataDeposit(_data) => Err(TranslationError::Unspendable),
+                TxOutput::AnyoneCanTake(_) => Err(TranslationError::Unspendable),
+            },
+            InputInfo::Account { outpoint } => match outpoint.account() {
+                AccountSpending::DelegationBalance(delegation_id, _amount) => {
+                    let dest = ctx
+                        .get_delegation_spend_destination(delegation_id)?
+                        .ok_or(TranslationError::DelegationNotFound(*delegation_id))?;
+                    Ok(checksig(&dest))
+                }
+            },
+            InputInfo::AccountCommand { command } => match command {
+                AccountCommand::MintTokens(token_id, _)
+                | AccountCommand::UnmintTokens(token_id)
+                | AccountCommand::LockTokenSupply(token_id)
+                | AccountCommand::FreezeToken(token_id, _)
+                | AccountCommand::UnfreezeToken(token_id)
+                | AccountCommand::ChangeTokenAuthority(token_id, _) => {
+                    let dest = ctx
+                        .get_tokens_authority(token_id)?
+                        .ok_or(TranslationError::TokenNotFound(*token_id))?;
+                    Ok(checksig(&dest))
+                }
+                AccountCommand::ConcludeOrder(order_id) => {
+                    let dest = ctx
+                        .get_orders_conclude_destination(order_id)?
+                        .ok_or(TranslationError::OrderNotFound(*order_id))?;
+                    Ok(checksig(&dest))
+                }
+                AccountCommand::FillOrder(_, _, _) => Ok(WitnessScript::TRUE),
+            },
+        }
+    }
+}
+
+pub struct SignatureOnlyReward;
+
+impl<C: SignatureInfoProvider> TranslateInput<C> for SignatureOnlyReward {
+    fn translate_input(_ctx: &C) -> Result<WitnessScript, TranslationError> {
+        // Not used anywhere.
+        // But it's important to outline that if needed the reward implementation must be different
+        // because staking/decommissioning destinations are not the same.
+        unimplemented!()
+    }
+}
```

Comment:
Why keep it if it's unused?

Reply:
I wanted to make it explicit for the future. So that noone tried to use implementation from transaction with reward (it might be a problem because both implement Transactable)